superset-frontend/src/cccs-viz/plugins/plugin-chart-at-a-glance-user-id/src/AtAGlanceUserID.tsx

```diff
@@ -0,0 +1,268 @@
+    i++
+  ) {
+    const filter = initialFormData.formData.extraFormData.filters[i];
+    if (filter.col === 'user_id') {
+      const localuserID: string = filter.val[0];
+      if (localuserID !== userIDString) {
+        setAadDataManger({
+          ...aadDataManager,
+          data: initialFormData.data,
+          formData: initialFormData,
+        });
+        setUserIDString(localuserID);
+        hasFiltered = false;
+      }
+      break;
+    }
+  }
+
+  if (
+    !aadDataManager.isLoading ||
+    (aadDataManager.isInit && hasFiltered === false)
+  ) {
+    canadianIpsList = aadDataManager.data.filter(function (item) {
+      return (
+        getPayloadField('client_ip_cbs_geo_country_name', item) === 'canada' &&
+        getPayloadField('operation', item) === 'UserLoggedIn'
+      );
+    });
+
+    nonCanadianIpsList = aadDataManager.data.filter(function (item) {
+      return (
+        getPayloadField('client_ip_cbs_geo_country_name', item) !== 'canada' &&
+        getPayloadField('operation', item) === 'UserLoggedIn'
+      );
+    });
+
+    nonSucsessfulCanadianIpsList = aadDataManager.data.filter(function (item) {
+      return (
+        getPayloadField('client_ip_cbs_geo_country_name', item) === 'canada' &&
+        getPayloadField('operation', item) !== 'UserLoggedIn'
+      );
+    });
+
+    nonSucsessfulNonCanadianIpsList = aadDataManager.data.filter(function (
+      item,
+    ) {
+      return (
+        getPayloadField('client_ip_cbs_geo_country_name', item) !== 'canada' &&
+        getPayloadField('operation', item) !== 'UserLoggedIn'
+      );
+    });
+
+    hasFiltered = true;
+  }
+
+  return (
+    <>
+      <div>
+        <RiGlobalFill />
+        <span>At a Glance</span>
+      </div>
+      <div>
+        <table>
+          <tr>
+            <td>User Email: {userIDString}</td>
+          </tr>
+          <tr>
+            <td>
+              User ID:{' '}
+              {aadDataManager.isLoading
+                ? 'Loading'
+                : getPayloadField('user_key', aadDataManager.data[0])}
+            </td>
+          </tr>
+        </table>
+      </div>
+      <div>
+        <table>
+          <tr>
+            <Collapse bordered expandIconPosition="left" ghost>
+              <Collapse.Panel
+                header={
+                  <span className="header">
+                    {' '}
+                    Number of Successful Canadian Login Attempts :{' '}
+                    {aadDataManager.isLoading
+                      ? 'Loading'
+                      : canadianIpsList.length}{' '}
+                  </span>
+                }
+                key="1"
+              >
+                {aadDataManager.isLoading && !aadDataManager.isInit ? (
+                  <></>
+                ) : (
+                  <ul>
+                    {canadianIpsList.map((a: { client_ip: string }) => (
+                      <li>
+                        {/* TODO: Add link back in
```

Comment:
Ticket reference for this? (Presumably the same one as the earlier TODO?)

Reply:
https://cccs.atlassian.net/browse/CLDN-1258

Reply:
Sweet! In CLSM, we would do something like "TODO: CLDN-1258 Add link back in". So people looking at the code would know where to go look if they wanted to see details about what was to be done, and/or if this ticket had gotten closed and this TODO got missed.